forc-plugins/forc-client/src/op/call/mod.rs

```diff
@@ -234,21 +242,56 @@ pub async fn call(cmd: cmd::Call) -> anyhow::Result<String> {
         }
     };
 
-    forc_tracing::println_action_green("receipts:", &format!("{:#?}", receipts));
-    forc_tracing::println_action_green("tx hash:", &tx_hash.to_string());
-    forc_tracing::println_action_green("result:", &result);
+    // print receipts
+    if show_receipts {
+        forc_tracing::println_action_green("receipts:", &format!("{:#?}", receipts));
+    }
+
+    // decode logs
+    let program_abi = sway_core::asm_generation::ProgramABI::Fuel(parsed_abi);
+    let logs = receipts
+        .iter()
+        .filter_map(|receipt| {
+            if let Receipt::LogData {
+                rb,
+                data: Some(data),
+                ..
+            } = receipt
+            {
+                return forc_test::decode_log_data(&rb.to_string(), data, &program_abi)
+                    .ok()
+                    .map(|decoded| decoded.value);
+            }
+            None
+        })
+        .collect::<Vec<_>>();
+
+    // print logs
+    if !logs.is_empty() {
+        forc_tracing::println_green_bold("logs:");
+        logs.iter()
+            .for_each(|log| forc_tracing::println_yellow(&format!("  {:#}", log)));
```

Comment:
hmm maybe white is better here?

Reply:
I fully agree, I think functional style calls are great especially at the places we convert types to each other but around places we print stuff or do stuff with side-effects in more general, they are not very easy to debug. 

Once something starts happening as you need to convert into more imperative style, or need to add braces etc just to print/debug stuff around it. This might be because i generally do not use debugger if there is no explicit need to use one :)

![image](https://github.com/user-attachments/assets/05cf10f5-4e59-4686-a7c4-400bebf595a7)

Reply:
[1127d29](https://github.com/FuelLabs/sway/pull/6968/commits/1127d293879651c4af8f7d655585bfb03f465a2a)